src/BookNavigator/volumes/volumes-provider.js

```diff
@@ -17,84 +13,71 @@ export default class VolumesProvider {
+    this.bookreader = bookreader;
     this.onProviderChange = onProviderChange;
-    this.component = document.createElement("viewable-files");
+    this.baseHost = baseHost;
 
     const files = bookreader.options.multipleBooksList.by_subprefix;
     this.viewableFiles = Object.keys(files).map(item => files[item]);
     this.volumeCount = Object.keys(files).length;
 
-    /** @type {import('../../BookReader').default} */
-    this.bookreader = bookreader;
-
-    this.component.subPrefix = bookreader.options.subPrefix || "";
-    this.component.hostUrl = baseHost;
-    this.component.viewableFiles = this.viewableFiles;
-
     this.id = "volumes";
     this.label = `Viewable files (${this.volumeCount})`;
-    this.icon = html`${volumesIcon}`;
+    this.icon = html`${viewableFilesIcon}`;
+    this.sortOrderBy = sortOptions.default;
+
+    this.component = document.createElement("iaux-viewable-files");
+    this.component.addSortToUrl = true;
+    this.component.subPrefix = bookreader.options.subPrefix || "";
+    this.component.baseHost = baseHost;
+    this.component.fileList = [...this.viewableFiles];
 
-    this.sortOrderBy = sortType.default;
+    this.sortFilesComponent = document.createElement("iaux-sort-viewable-files");
+    this.sortFilesComponent.fileListRaw = this.viewableFiles;
+    this.sortFilesComponent.addEventListener('fileListSorted', (e) => this.handleFileListSorted(e));
+    this.actionButton = this.sortFilesComponent;
 
     // get sort state from query param
     if (this.bookreader.urlPlugin) {
       this.bookreader.urlPlugin.pullFromAddressBar();
 
       const urlSortValue = this.bookreader.urlPlugin.getUrlParam('sort');
-      if (urlSortValue === sortType.title_asc || urlSortValue === sortType.title_desc) {
+      if (urlSortValue === sortOptions.title_asc || urlSortValue === sortOptions.title_desc) {
         this.sortOrderBy = urlSortValue;
       }
     }
-    this.sortVolumes(this.sortOrderBy);
-  }
 
-  get sortButton() {
-    const sortIcons = {
-      default: html`
-        <button class="sort-by neutral-icon" aria-label="Sort volumes in initial order" @click=${() => this.sortVolumes("title_asc")}>${sortNeutralIcon}</button>
-      `,
-      title_asc: html`
-        <button class="sort-by asc-icon" aria-label="Sort volumes in ascending order" @click=${() => this.sortVolumes("title_desc")}>${sortAscIcon}</button>
-      `,
-      title_desc: html`
-        <button class="sort-by desc-icon" aria-label="Sort volumes in descending order" @click=${() => this.sortVolumes("default")}>${sortDescIcon}</button>
-      `,
-    };
-
-    return sortIcons[this.sortOrderBy];
+    this.sortFilesComponent.sortVolumes(this.sortOrderBy);
+
+    this.onProviderChange(this.bookreader);
   }
 
   /**
    * @param {'default' | 'title_asc' | 'title_desc'} sortByType
    */
-  sortVolumes(sortByType) {
-    let sortedFiles = [];
+  async handleFileListSorted(event) {
+    const { sortType, sortedFiles } = event.detail;
```

Comment:
can we make the naming consistent? the renaming of the variable `sortOptions` is better, so... in this case, can we make it `sortOption` instead?

Reply:
thanks, ive updated the file to use `sortTypes` and then added a new type to help make sense of usage around file